### PR TITLE
fix: fixed color selection bug when no preset is selected

### PR DESCRIPTION
**Pull request by mikasackermn**
There was a problem in Custom Color, that we couldn't change a color from Custom Color when a color was not selected from the presets, this problem has been solved in this PR.

### CODE REVIEW

**Comment by samobasquiat** [comment]
These are not really related to this PR, but it would be better to fix them:
. In package.json, there are two "react-color"
. There are two files named ColorPickerstyles and ColorPicker.styles in the ColorPicker component. One of them should be removed.

File: widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx [hunk 1]
```diff
@@ -175,14 +218,12 @@ export function CustomColorsSection(props: CustomColorsTypes) {
                 placeholder={widgetColor.label}
                 color={getMainColor(widgetColor.key, tab, theme) || ''}
                 onChangeColor={(color) =>
-                  onChangeColors(widgetColor.key, tab, color)
+                  onChangeColor(widgetColor.key, tab, color)
                 }
                 onReset={() => onResetColor(widgetColor.key, tab)}
                 resetDisable={
```
Comment (samobasquiat) [inline widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx:224]: https://github.com/rango-exchange/rango-client/assets/123297941/3af7cf0f-2050-4a0f-978b-f8cd95052d5a

When no preset is selected, the reset button doesn't work. I think it worked in the previous version.

File: widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx [hunk 2]
```diff
@@ -70,9 +88,38 @@ export function CustomColorsSection(props: CustomColorsTypes) {
...
+  }, [theme?.colors]);
+
+  const onChangeColor = (
+    name: WidgetColorsKeys,
+    mode: 'light' | 'dark',
+    color?: string
+  ) => {
+    let resetColors = false;
+    if (selectedColors.tab !== tab) {
+      resetColors = true;
```
Comment (samobasquiat) [inline widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx:108]: `resetColors = selectedColors.tab !== tab `is not better?

File: widget/playground/src/store/config.ts [hunk 3]
```diff
@@ -206,18 +194,34 @@ export const useConfigStore = createSelectors(
...
-              state.config.theme.colors = {
-                ...state.config.theme.colors,
-                [mode]: {
-                  ...state?.config?.theme.colors[mode],
-                  [name]: color,
-                },
-              };
+            if (state.config?.theme?.colors) {
+              let themes = { ...state.config.theme, singleTheme };
+              if (resetColors) {
```
Comment (samobasquiat) [inline widget/playground/src/store/config.ts:201]: Please add a comment in this section to improve code clarity.

File: widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx [hunk 4]
```diff
@@ -44,18 +47,33 @@ const Colors = (props: { mainColor?: string; secondColor?: string }) => {
 };
 
 export function CustomColorsSection(props: CustomColorsTypes) {
```
Comment (samobasquiat) [inline widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx:49]: 'selectedPresets' has been removed from the props but it is still present in the types.

File: widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx (hunk 1 above)
Comment (mikasackermn) [reply widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx:224]: fixed

File: widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx (hunk 2 above)
Comment (mikasackermn) [reply widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx:108]: yes, is better

File: widget/playground/src/store/config.ts (hunk 3 above)
Comment (mikasackermn) [reply widget/playground/src/store/config.ts:201]: I added

File: widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx (hunk 4 above)
Comment (mikasackermn) [reply widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx:49]: I used this prop

**Review by samobasquiat — APPROVED** [review]

File: widget/playground/src/constants/presets.ts [hunk 5]
```diff
@@ -248,3 +248,19 @@ export const WIDGET_COLORS: { key: WidgetColorsKeys; label: string }[] = [
...
+    foreground: undefined,
+  },
+  light: {
+    primary: undefined,
+    secondary: undefined,
+    neutral: undefined,
+    background: undefined,
+    foreground: undefined,
+  },
+};
```
Comment (RanGojo) [inline widget/playground/src/constants/presets.ts:266]: don't we need info property here?

File: widget/playground/src/store/config.ts [hunk 6]
```diff
@@ -206,18 +194,35 @@ export const useConfigStore = createSelectors(
...
-              state.config.theme.colors = {
-                ...state.config.theme.colors,
-                [mode]: {
-                  ...state?.config?.theme.colors[mode],
-                  [name]: color,
-                },
-              };
+            if (state.config?.theme?.colors) {
+              let themes = { ...state.config.theme, singleTheme };
+              // If the resetColors is true, all the colors should return to the default state, because the colors are changing in the new tab.
```
Comment (RanGojo) [inline widget/playground/src/store/config.ts:201]: should reset or should revert.

File: widget/playground/src/components/ColorPicker/ColorPicker.styles.ts [hunk 7]
```diff
@@ -1,10 +1,13 @@
...
 
 export const Container = styled('div', {
+  display: 'flex',
+  justifyContent: 'space-around',
+  alignItems: 'center',
   position: 'relative',
 });
 
-export const Color = styled('div', {
+export const ColorDiv = styled('div', {
```
Comment (RanGojo) [inline widget/playground/src/components/ColorPicker/ColorPicker.styles.ts:10]: We usually don't use this naming convention. maybe use ColorContainer or something like that.

File: widget/playground/src/constants/presets.ts (hunk 5 above)
Comment (mikasackermn) [reply widget/playground/src/constants/presets.ts:266]: yes, we do (fixed)

File: widget/playground/src/store/config.ts (hunk 6 above)
Comment (mikasackermn) [reply widget/playground/src/store/config.ts:201]: fixed

File: widget/playground/src/components/ColorPicker/ColorPicker.styles.ts (hunk 7 above)
Comment (mikasackermn) [reply widget/playground/src/components/ColorPicker/ColorPicker.styles.ts:10]: fixed

**Pull request merged**